all/projectAlcazar/src/sys/system.c

```diff
@@ -18,28 +18,70 @@ void enablePeripheral(ePeripherals peripheral, eEnableState enable)
 	{
 		pRCCRegister = &RCC_AHBENR;
 	}
-	else if(ePerif_TIM1 == peripheral || ePerif_TIM15 == peripheral ||
-			ePerif_SPI1EN == peripheral)
+	
+	if(enable == eEnabled)
 	{
-		pRCCRegister = &RCC_APB2ENR;
+		*pRCCRegister |= 1 << peripheral;
 	}
-	/*else if(ePerif_USART2EN == peripheral)
+	else
 	{
-		pRCCRegister = &RCC_APB1ENR;
+		*pRCCRegister &= ~(1 << peripheral);
+	}
+}
+
+
+void enablePeripheralRCC_APB1ENR(ePeripheralsRCC_APB1ENR peripheral, eEnableState enable)
+{
+	volatile unsigned int *pRCCRegister2;
+
+	if(ePerif_USART2EN == peripheral)
+	{
+		pRCCRegister2 = &RCC_APB1ENR;
+	}
+	else if(ePerif_TIM3EN == peripheral)
+	{
+		pRCCRegister2 = &RCC_APB1ENR;
 	}
-	
-	*/
 
 	if(enable == eEnabled)
 	{
-		*pRCCRegister |= 1 << peripheral;
+		*pRCCRegister2 |= 1 << peripheral;
 	}
 	else
 	{
-		*pRCCRegister &= ~(1 << peripheral);
+		*pRCCRegister2 &= ~(1 << peripheral);
+	}
+}
+
+
+void enablePeripheralRCC_APB2ENR(ePeripheralsRCC_APB2ENR peripheral, eEnableState enable)
+{
+	volatile unsigned int *pRCCRegister3;
+
+	if(ePerif_TIM1 == peripheral)
+	{
+		pRCCRegister3 = &RCC_APB1ENR;
```

Comment:
register should be RCC_APB2ENR not RCC_APB1ENR

Reply:
This fixed the freezing.